Labyrinth/src/Physics/Board.java

```diff
@@ -143,6 +147,21 @@ public Tile insert(Tile t, Point location, int direction){
 		return result;
 	}
 	
+	/**
+	 * Finds the furthest tile from the cursorLocation in the direction insertDirection.
+	 * @param insertDirection
+	 * @param cursorLocation
+	 * @return
+	 */
+	public Point getFurthestTile(int insertDirection, Point cursorLocation) {
+		Point newCoords = Directions.move(cursorLocation,insertDirection);
+		Tile nextTile = this.tileAt(newCoords);
+		if(nextTile != null) {
+			return this.getFurthestTile(insertDirection, newCoords);
+		}
+		return cursorLocation;
```

Comment:
For standardization:

Usually, recursive methods are written:
```
methodCall(){
  exitCase1
  exitCase2
  main logic in the method with recursive call
}
```
I think this is so you can easily verify that the method ends.

To help read this, can we swap the exit case `return cursorLocation` and the recursive call? We'll have to fix the if statement too.